### PR TITLE
Propose first very simple version of the website for 2.0

### PR DESCRIPTION
**Pull request by parisk**
Propose the very first version of xtermjs.org for 2.0 release of xterm.js based on Jekyll.

This website includes:

- docs
- blog

Preview at https://d8q3uus5.apps.lair.io/

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[*]
+indent_style = space
+indent_width = 2
+indent_size = 2
+insert_final_new_line = true
```
Comment (Tyriar) [inline .editorconfig]: `trim_trailing_whitespace = true`?

File: _config.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+defaults:
+  - {scope: {type: docs}, values: {layout: docs}}
+  - {scope: {type: posts}, values: {layout: post, permalink: '/blog/articles/:year-:month-:day-:title'}}
+
+include:
+  - _pages
+
+collections:
+    docs:
```
Comment (Tyriar) [inline _config.yml]: 2 space indent

File: _docs/api/Terminal.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+
+## Methods
+
+### `attachCustomKeydownHandler(customKeydownHandler)`
+
+- `customKeydownHandler` - Function - The custom KeyboardEvent handler to attach.
+
+Run a custom keydown handler before any keys are processed and
+allow consumers to decide what keys should be processed by the terminal.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: I'd word this:

> Attach a custom keydown handler to allow consumers to allow certain key strokes to skip processing by the terminal.

File: _docs/api/Terminal.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+```javascript
+// Log the keyCode of every keyDown event
+term.textarea.onkeydown = function (e) {
+  console.log('User pressed key with keyCode: ', e.keyCode);
+}
+```
+
+## Events
+
+The `Terminal` instances emit the following events.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: - Should there be general advice on how to listen for events?
- Do we need to expose all these events?
- The events section should go below methods imo

File: _docs/api/Terminal.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+Emitted when the terminal handles a keydown or keypress event.
+
+### `keydown`
+
+- `e` - KeyboardEvent - The original `keyDown`
+
+Emitted after a `keyDown` event on the terminal.
+
+### `keydown`
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: `keypress`

File: _docs/api/Terminal.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+- `e` - KeyboardEvent - The original `keyPress`
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: keyPress -> keypress

File: _docs/api/Terminal.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+
+Emitted after a `keyPress` event on the terminal.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: keyPress -> keypress

File: _docs/api/Terminal.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+
+### `attachCustomKeydownHandler(customKeydownHandler)`
+
+- `customKeydownHandler` - Function - The custom KeyboardEvent handler to attach.
+
+Run a custom keydown handler before any keys are processed and
+allow consumers to decide what keys should be processed by the terminal.
+
+```javascript
+// Completely ignore the `Esc` key using a custom keydown handler.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: Tab might be a better example as it was the critical use case for me to ensure accessibility.

File: _docs/api/Terminal.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+    // Do nothing
+    e.preventDefault();
+    return false;
+  }
+});
+```
+
+### `blur`
+
+Remove the focus from the terminal
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: Full stop

File: _docs/api/Terminal.md [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+- `callback` - Function - The callback to stop running on the event
+
+Stop running a callback on an event.
+
+```javascript
+// Stop logging the terminal's size when it gets resized.
+term.off('resize', logResize)
+```
+
+### `emit(event, data)`
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: :-1: to exposing this

File: _docs/api/Terminal.md [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+
+// Expose the terminal into `terminalParent`.
+term.open(terminalParent);
+```
+
+### `refresh(start, end, queue)`
+
+- `start` - Number - The first row to be refreshed
+- `end` - Number - The last row to be refreshed
+- `queue` - Boolean - Queue the refresh to run asynchronously, when it's more optimal
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: Do we need to expose the `queue` parameter?

File: _docs/api/Terminal.md [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+```javascript
+var terminalParent = document.getElementById('xterm-container');
+
+// Refresh the contents of the terminal from the 10th to the 20th line.
+term.refresh(10, 20);
+```
+
+### `reset()`
+
+Reset the terminal; reconstruct the instance and re-render the whole screen.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: screen -> buffer, screen implies outside the Terminal.element

File: _docs/guides/download.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+## Bower
+Another way to download xterm.js for your project is through [Bower](http://bower.io/).
+
+```
+bower install --save xterm.js
+```
+
+## GitHub Releases
+All xterm.js releases are available as tar and zip archives at [https://github.com/sourcelair/xterm.js/releases](https://github.com/sourcelair/xterm.js/releases), to download them manually.
```
Comment (Tyriar) [inline _docs/guides/download.md]:
```
All xterm.js releases are available as tar and zip archives via [GitHub](https://github.com/sourcelair/xterm.js/releases).
```

Also maybe clarify the tar compression (tar.gz, tar.xz) or refer to as tarball?

File: _docs/guides/download.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+## Clone the Git repository
+You can always clone the whole xterm.js repository and checkout any arbitrary commit you would like to use. We suggest not doing this though unless you are certain about your actions, as only official tagged releases of xterm.js are considered production-ready.
```
Comment (Tyriar) [inline _docs/guides/download.md]: Does this section need to be included at all?

File: _pages/index.html [hunk 15]
```diff
@@ -0,0 +1,13 @@
+---
+layout: default
+permalink: /
+---
+
+<div class="home">
+
+  <article>
+  	{% capture about %}{% include xtermjs-description.md %}{% endcapture %}
```
Comment (Tyriar) [inline _pages/index.html]: Can we include a barebones example of the JS on the home page?

File: _docs/api/Terminal.md (hunk 4 above)
Comment (parisk) [inline _docs/api/Terminal.md]: > Should there be general advice on how to listen for events?

👍  adding a link to the `on` and `off` methods for now with a brief explanation.

> Do we need to expose all these events?

I think that all of them are useful in the public API. About which of them are you skeptical?

> The events section should go below methods imo

Moving.

File: _docs/api/Terminal.md (hunk 10 above)
Comment (parisk) [inline _docs/api/Terminal.md]: Indeed. Bad idea.

File: _docs/api/Terminal.md (hunk 11 above)
Comment (parisk) [inline _docs/api/Terminal.md]: I think we should. I guess that depending on the use case, each consumer can use their own option.

File: _docs/guides/download.md (hunk 14 above)
Comment (parisk) [inline _docs/guides/download.md]: Mostly to point out that this should not be used for production.

File: _pages/index.html (hunk 15 above)
Comment (parisk) [inline _pages/index.html]: What would you suggest? I was thinking implementing a dumb terminal for the visitors to play with, in the home page but needed some effort.

**Comment by parisk** [comment]
@Tyriar fixed most of your comments. Feel free to take a look again, when it's OK for you.

File: _docs/api/Terminal.md (hunk 4 above)
Comment (Tyriar) [inline _docs/api/Terminal.md]: I don't think I touch `emit` at all in my implementation so it's non-essential for me

Comment (Tyriar) [inline _docs/api/Terminal.md]: I only use `data` is why I made that comment, haven't had a need for the others.

File: _docs/api/Terminal.md (hunk 11 above)
Comment (Tyriar) [inline _docs/api/Terminal.md]: We might get rid of queue as part of https://github.com/sourcelair/xterm.js/issues/280, plus it should only be relied upon internally unless the need arises imo

File: _pages/index.html (hunk 15 above)
Comment (Tyriar) [inline _pages/index.html]: Just some minimal example if we can get it small enough to demonstrate that it only takes x lines to get a usable terminal in your web app. Something like this:

```js
var terminal = xterm();
terminal.open(document.querySelector('#terminal'));
var process = // init process...
process.on('data', terminal.write);
terminal.on('data', function (data) {
	process.send({
		event: 'input',
		data: data
	});
	return false;
});
```

:+1: to dumb terminal as well, interactive examples are great.

File: _docs/api/Terminal.md (hunk 11 above)
Comment (parisk) [inline _docs/api/Terminal.md]: What about dropping `refresh` from the public API? Sounds weird to document two out of three arguments 😅 . Or do you propose some kind of refactoring here before the release?

File: _docs/api/Terminal.md (hunk 4 above)
Comment (parisk) [inline _docs/api/Terminal.md]: We use most of the events there, so let's leave them there. They can help other people also build their own add-ons.

**Comment by parisk** [comment]
I added also a "Quick start" example in the bottom of the landing page.

File: _docs/api/Terminal.md (hunk 11 above)
Comment (Tyriar) [inline _docs/api/Terminal.md]: Actually now that you mention it, the only place where I need it is immediately after what is now `setOption`:

```ts
	public setCursorBlink(blink: boolean): void {
		if (this._xterm && this._xterm.getOption('cursorBlink') !== blink) {
			this._xterm.setOption('cursorBlink', blink);
			this._xterm.refresh(0, this._xterm.rows - 1);
		}
	}
```

So yeah, let's remove it until there is a solid use case! :+1:

File: _docs/api/Terminal.md [hunk 16]
```diff
@@ -0,0 +1,286 @@
...
+
+Emitted when the terminal gets a content refresh (re-render).
+
+### `resize`
+
+- `data` - Object - `{terminal: this, cols: x, rows: y}`
+
+Emitted when the terminal gets resized to a new geometry.
+
+### `scrollDisp`
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: Isn't this event called `scroll`?

File: _pages/index.html [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    <head>
+      <link rel="stylesheet" href="bower_components/xterm.js/dist/xterm.css" />
+      <script src="bower_components/xterm.js/dist/xterm.js"></script>
+    </head>
+    <body>
+      <div id="terminal"></div>
+      <script>
+      	var term = new Terminal();
+        term.open(document.getElementById('#terminal'));
+        term.write('Hello from xterm.js $ ')
```
Comment (Tyriar) [inline _pages/index.html]: How about something like this to make it pop a bit more: `Hello from \033[1;3;31mxterm.js\033[0m $ `

**Pull request merged**